### PR TITLE
Payment

### PR DESCRIPTION
**Pull request by SerhiiDielieu**
Added tests for payment/init
and payment/pin
Refactoring

### CODE REVIEW

**Comment by SerhiiDielieu** [comment]
@adamsabel please review

File: src/main/java/payment/EndPointsPayment.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public enum EndPointsPayment {
+
+    INIT_PAYMENT("/payment/init"),
+    RETRY_IRIS_PAYMENT("/payment/iris/retry"),
+    PIN_PAYMENT("/payment/pin"),
+    CANCEL_PAYMENT("/payment/cancel");
+
+    private final String url;
+
+    public String getUrl() {
```
Comment (adamsabel) [inline src/main/java/payment/EndPointsPayment.java]: use @ Getter

File: src/main/java/payment/InitStatusResponses.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public enum InitStatusResponses {
+    FAILED_IRIS_NOT_MATCHED("FAILED_IRIS_NOT_MATCHED"),
+    AUTHORIZED_IRIS_MATCHED("AUTHORIZED_IRIS_MATCHED"),
+    PIN_NEEDED("PIN_NEEDED");
+
+
+    private final String message;
+
+    public String getMessage() {
+        return message;
```
Comment (adamsabel) [inline src/main/java/payment/InitStatusResponses.java]: Use @ getttttterrrrrrrrrrrrrrrrr

File: src/main/java/payment/PinStatusResponses.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public enum PinStatusResponses {
+    AUTHORIZED_PIN_MATCHED("AUTHORIZED_PIN_MATCHED"),
+    FAILED_PIN_NOT_MATCHED("FAILED_PIN_NOT_MATCHED"),
+    TERMINATED_PIN_AUTH_LAST_RETRY("TERMINATED_PIN_AUTH_LAST_RETRY");
+
+
+    private final String message;
+
+    public String getMessage() {
+        return message;
```
Comment (adamsabel) [inline src/main/java/payment/PinStatusResponses.java]: Use @ gettttteerrrrrrrrrrrrrr

File: src/main/java/phone/PhoneNumberDB.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    User_542971061("542971061"),
+    User_759309272("759309272"),
+    User_482779939("482779939"),
+    User_464734812("464734812"),
+    User_656256042("656256042");
+
+    private final String phoneNumber;
+
+    public String getPhoneNumber() {
+        return phoneNumber;
```
Comment (adamsabel) [inline src/main/java/phone/PhoneNumberDB.java]: Use @ gettttttttterrrrrrr

File: src/main/java/util/ExecuteDBqueryUtil.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package util;
+
+import java.sql.*;
+
+public class ExecuteDBqueryUtil {
+
+    public ExecuteDBqueryUtil() {
```
Comment (adamsabel) [inline src/main/java/util/ExecuteDBqueryUtil.java]: ITS UTIL CLASS BOY

File: src/main/java/util/PropertiesUtil.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        while (true) {
+            try {
+                if ((line = br.readLine()) == null) break;
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            String[] values = line.split(",");
+            int[] intValues = new int[values.length];
+            for (int i = 0; i < values.length; i++) {
```
Comment (adamsabel) [inline src/main/java/util/PropertiesUtil.java]: Use foreach

File: src/main/java/util/PropertiesUtil.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                } catch (NumberFormatException nfe) {
+                    System.out.println(intValues[i]);
+                    continue;
+                }
+            }
+
+            result = new byte[intValues.length];
+
+            for (int k = 0; k < result.length; k++) {
+                result[k] = (byte) intValues[k];
```
Comment (adamsabel) [inline src/main/java/util/PropertiesUtil.java]: use foreach

File: src/main/java/payment/EndPointsPayment.java (hunk 1 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/EndPointsPayment.java]: done

File: src/main/java/payment/InitStatusResponses.java (hunk 2 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/InitStatusResponses.java]: done

File: src/main/java/payment/PinStatusResponses.java (hunk 3 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/PinStatusResponses.java]: done

File: src/main/java/phone/PhoneNumberDB.java (hunk 4 above)
Comment (SerhiiDielieu) [reply src/main/java/phone/PhoneNumberDB.java]: done

File: src/main/java/util/ExecuteDBqueryUtil.java (hunk 5 above)
Comment (SerhiiDielieu) [reply src/main/java/util/ExecuteDBqueryUtil.java]: constructor changed to private

File: src/main/java/util/PropertiesUtil.java (hunk 6 above)
Comment (SerhiiDielieu) [reply src/main/java/util/PropertiesUtil.java]: fixed

File: src/main/java/util/PropertiesUtil.java (hunk 7 above)
Comment (SerhiiDielieu) [reply src/main/java/util/PropertiesUtil.java]: fixed

**Comment by SerhiiDielieu** [comment]
@adamsabel

File: src/main/java/payment/EndPointsPayment.java (hunk 1 above)
Comment (adamsabel) [inline src/main/java/payment/EndPointsPayment.java]: Use getter

File: src/main/java/payment/InitStatusResponses.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+@AllArgsConstructor
+public enum InitStatusResponses {
+    FAILED_IRIS_NOT_MATCHED("FAILED_IRIS_NOT_MATCHED"),
+    AUTHORIZED_IRIS_MATCHED("AUTHORIZED_IRIS_MATCHED"),
+    PIN_NEEDED("PIN_NEEDED");
+
+
+    private final String message;
+
+    public String getMessage() {
```
Comment (adamsabel) [inline src/main/java/payment/InitStatusResponses.java]: Use getter

File: src/main/java/payment/PinStatusResponses.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+@AllArgsConstructor
+public enum PinStatusResponses {
+    AUTHORIZED_PIN_MATCHED("AUTHORIZED_PIN_MATCHED"),
+    FAILED_PIN_NOT_MATCHED("FAILED_PIN_NOT_MATCHED"),
+    TERMINATED_PIN_AUTH_LAST_RETRY("TERMINATED_PIN_AUTH_LAST_RETRY");
+
+
+    private final String message;
+
+    public String getMessage() {
```
Comment (adamsabel) [inline src/main/java/payment/PinStatusResponses.java]: Use getter

File: src/main/java/payment/PinStatusResponses.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package payment;
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
```
Comment (adamsabel) [inline src/main/java/payment/PinStatusResponses.java]: Constructos in enums are private

File: src/main/java/phone/PhoneNumberDB.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+public enum PhoneNumberDB {
+    User_542971061("542971061"),
+    User_759309272("759309272"),
+    User_482779939("482779939"),
+    User_464734812("464734812"),
+    User_656256042("656256042");
+
+    private final String phoneNumber;
+
+    public String getPhoneNumber() {
```
Comment (adamsabel) [inline src/main/java/phone/PhoneNumberDB.java]: Getterr

File: src/main/java/phone/PhoneNumberDB.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+package phone;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public enum PhoneNumberDB {
```
Comment (adamsabel) [inline src/main/java/phone/PhoneNumberDB.java]: Why did you create this enum?

File: src/main/java/util/ExecuteDBqueryUtil.java (hunk 5 above)
Comment (adamsabel) [inline src/main/java/util/ExecuteDBqueryUtil.java]: Its util class

File: src/main/java/util/ExecuteDBqueryUtil.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public static String getQueryResult(String query, String column) {
```
Comment (adamsabel) [inline src/main/java/util/ExecuteDBqueryUtil.java]: This query returns only one result, please specify it somehow

File: src/main/java/util/InitPaymentUtil.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.io.FileNotFoundException;
+
+import static groovy.json.JsonOutput.toJson;
+import static payment.EndPointsPayment.INIT_PAYMENT;
+import static payment.EndPointsPayment.PIN_PAYMENT;
+import static util.GenerateBodyForRequestUtil.whenSendPost;
+import static util.PropertiesUtil.readFile;
+
+public class InitPaymentUtil {
```
Comment (adamsabel) [inline src/main/java/util/InitPaymentUtil.java]: private constructor

File: src/main/java/util/InitPaymentUtil.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    public static Response initAndSendPinForPayment(String amount, String currency, String pin, String walletPhoneNumber, String file){
+    InitPaymentRequest paymentRequest = null;
+            try {
```
Comment (adamsabel) [inline src/main/java/util/InitPaymentUtil.java]: Formatting

File: src/main/java/util/InitPaymentUtil.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+        e.printStackTrace();
+    }
+    Response responseInit = whenSendPost(INIT_PAYMENT.getUrl(), toJson(paymentRequest));
+    ResponseBody body = responseInit.getBody();
+    String id = body.path("paymentId").toString();
+
+    PinMatchPaymentRequest pinMatchPaymentRequest = PinMatchPaymentRequest.builder().pin(pin).phoneNumber(walletPhoneNumber).id(id).build();
+    return  whenSendPost(PIN_PAYMENT.getUrl(), toJson(pinMatchPaymentRequest));
+
+}}
```
Comment (adamsabel) [inline src/main/java/util/InitPaymentUtil.java]: Idk if this is util class, this method does too many things as to be util. You can split it into two separate methods. Util classes serve purpouse of making life easier and not performing business stuff

File: src/test/java/payment/InitPaymentTest.java [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+    @Parameters("https://payeye.atlassian.net/browse/BACKEND-80")
+    @Test(invocationCount = 1, description = "/payment/init")
+    public void shouldAcceptRegisteredUserWithCorrectTemplateAndLongAmountAndPLNCurrency() {
+        List<String> amounts = List.of("10.10", "0.10", "0.01", "01", "49.99");
+        String currency = "PLN";
+
+        String phoneNumber = generatePhoneNumber();
+
+        try {
+            addToWalletAndEnrollNumber(phoneNumber, readFile("src/test/template.txt"), null, null);
```
Comment (adamsabel) [inline src/test/java/payment/InitPaymentTest.java]: readFile should consume the exception

File: src/main/java/payment/EndPointsPayment.java (hunk 1 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/EndPointsPayment.java]: done

File: src/main/java/payment/InitStatusResponses.java (hunk 8 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/InitStatusResponses.java]: done

File: src/main/java/payment/PinStatusResponses.java (hunk 9 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/PinStatusResponses.java]: done

File: src/main/java/payment/PinStatusResponses.java (hunk 10 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/PinStatusResponses.java]: done

File: src/main/java/phone/PhoneNumberDB.java (hunk 11 above)
Comment (SerhiiDielieu) [reply src/main/java/phone/PhoneNumberDB.java]: done

File: src/main/java/phone/PhoneNumberDB.java (hunk 12 above)
Comment (SerhiiDielieu) [reply src/main/java/phone/PhoneNumberDB.java]: to store users that are already in DB

File: src/main/java/util/ExecuteDBqueryUtil.java (hunk 5 above)
Comment (SerhiiDielieu) [reply src/main/java/util/ExecuteDBqueryUtil.java]: done

File: src/main/java/util/ExecuteDBqueryUtil.java (hunk 13 above)
Comment (SerhiiDielieu) [reply src/main/java/util/ExecuteDBqueryUtil.java]: fixed

File: src/main/java/util/InitPaymentUtil.java (hunk 14 above)
Comment (SerhiiDielieu) [reply src/main/java/util/InitPaymentUtil.java]: fixed

File: src/main/java/util/InitPaymentUtil.java (hunk 15 above)
Comment (SerhiiDielieu) [reply src/main/java/util/InitPaymentUtil.java]: fixed

File: src/test/java/payment/InitPaymentTest.java (hunk 17 above)
Comment (SerhiiDielieu) [reply src/test/java/payment/InitPaymentTest.java]: ...

File: src/main/java/util/InitPaymentUtil.java (hunk 16 above)
Comment (SerhiiDielieu) [reply src/main/java/util/InitPaymentUtil.java]: reconfigured to method that initialize payment and return payment id

**Comment by SerhiiDielieu** [comment]
@adamsabel

File: src/main/java/payment/InitStatusResponses.java [hunk 18]
```diff
@@ -0,0 +1,16 @@
+package payment;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
```
Comment (adamsabel) [inline src/main/java/payment/InitStatusResponses.java]: Enums have private access constructor

File: src/main/java/util/InitPaymentUtil.java [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+import java.io.FileNotFoundException;
+
+import static groovy.json.JsonOutput.toJson;
+import static payment.EndPointsPayment.INIT_PAYMENT;
+import static payment.EndPointsPayment.PIN_PAYMENT;
+import static util.GenerateBodyForRequestUtil.whenSendPost;
+import static util.PropertiesUtil.readFile;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class InitPaymentUtil {
```
Comment (adamsabel) [inline src/main/java/util/InitPaymentUtil.java]: For util classes lombok does not support throwing exceptions, so you need to write constructor by hand

File: src/main/java/payment/InitStatusResponses.java (hunk 18 above)
Comment (SerhiiDielieu) [reply src/main/java/payment/InitStatusResponses.java]: fixed

File: src/main/java/util/InitPaymentUtil.java (hunk 19 above)
Comment (SerhiiDielieu) [reply src/main/java/util/InitPaymentUtil.java]: fixed

**Comment by SerhiiDielieu** [comment]
@adamsabel

File: pom.xml [hunk 20]
```diff
@@ -82,5 +82,11 @@
             <version>${restassured.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.2.8</version>
```
Comment (adamsabel) [inline pom.xml]: put into property

File: src/main/java/phone/PhoneNumberDB.java (hunk 12 above)
Comment (adamsabel) [reply src/main/java/phone/PhoneNumberDB.java]: What if DB goes down and you loose all the data inside? Your enum does not work then

File: pom.xml (hunk 20 above)
Comment (SerhiiDielieu) [reply pom.xml]: fixed

**Comment by SerhiiDielieu** [comment]
@adamsabel

**Pull request merged**